scipy/stats/_morestats.py

```diff
@@ -955,28 +978,39 @@ def boxcox_llf(lmb, data):
     >>> plt.show()
 
     """
+    # _axis_nan_policy decorator does not currently support these for non-NumPy arrays
+    kwargs = {}
+    if keepdims is not False:
+        kwargs[keepdims] = keepdims
+    if nan_policy != 'propagate':
+        kwargs[nan_policy] = nan_policy
+    return _boxcox_llf(data, lmb=lmb, axis=axis, **kwargs)
+
+
+@_axis_nan_policy_factory(lambda x: x, n_outputs=1, default_axis=0,
+                          result_to_tuple=lambda x: (x,))
+def _boxcox_llf(data, axis=0, *, lmb):
     xp = array_namespace(data)
     data = xp_promote(data, force_floating=True, xp=xp)
-
-    N = data.shape[0]
+    N = data.shape[axis]
     if N == 0:
-        return xp.nan
+        return _get_nan(data, xp=xp)
 
     logdata = xp.log(data)
 
     # Compute the variance of the transformed data.
     if lmb == 0:
-        logvar = xp.log(xp.var(logdata, axis=0))
+        logvar = xp.log(xp.var(logdata, axis=axis))
     else:
         # Transform without the constant offset 1/lmb.  The offset does
         # not affect the variance, and the subtraction of the offset can
         # lead to loss of precision.
         # Division by lmb can be factored out to enhance numerical stability.
         logx = lmb * logdata
-        logvar = _log_var(logx, xp) - 2 * math.log(abs(lmb))
+        logvar = _log_var(logx, xp, axis) - 2 * math.log(abs(lmb))
 
-    res = (lmb - 1) * xp.sum(logdata, axis=0) - N/2 * logvar
-    res = xp.astype(res, data.dtype, copy=False)
+    res = (lmb - 1) * xp.sum(logdata, axis=axis) - N/2 * logvar
+    res = xp.astype(res, data.dtype)
```

Comment:
I think we can merge as-is, but I'm wondering why remove `copy=False` here?

TBH, we can probably remove this line in a follow-up. It was probably due to old dtype manipulation of `sum`, but `array-api-compat` should be protecting us from that now.